src/AbstractAlgebra.jl

```diff
@@ -1238,13 +1238,26 @@ export Generic
 
 ###############################################################################
 #
-#   Polynomial Ring S, x = R["x"] syntax
+#   misc
+#
+###############################################################################
+
+include("misc/ProductIterator.jl")
+include("misc/Evaluate.jl")
+include("misc/VarNames.jl")
+
+###############################################################################
+#
+#   Polynomial Ring S, x = R[:x] syntax
 #
 ###############################################################################
 
 getindex(R::NCRing, s::VarName) = polynomial_ring(R, s)
-getindex(R::NCRing, s::VarName, ss::VarName...) =
-   polynomial_ring(R, [Symbol(x) for x in (s, ss...)])
+# For compatibility `R[:x, :y]` returns `S, [x, y]` instead of `S, x, y`
```

Comment:
"For compatibility" with what? Do you mean backwards compatibility, i.e. "because we always did it that way?

I'd just omit this. But a comment explaining what goes on still is helpful. So perhaps
```suggestion
# `R[:x, :y]` returns `S, [x, y]` instead of `S, x, y`
```

Reply:
Indeed, I would much prefer to return `S, x, y`, since that would be consistent (just call `polynomial_ring(R, s1, s...)`) and would simplify the common case of `S, (x, y) = R[:x, :y]` to `S, x, y = R[:x, :y]`. Though, I fear, that would break lots of code. That annoyed me, and I wrote that comment to cope with my annoyance.